src/go/rpk/pkg/cli/container/start.go

```diff
@@ -66,60 +68,51 @@ func newStartCommand() *cobra.Command {
 			// (POSIX standard)
 			UnknownFlags: true,
 		},
-		RunE: func(_ *cobra.Command, _ []string) error {
+		Run: func(cmd *cobra.Command, _ []string) {
 			if nodes < 1 {
-				return errors.New(
-					"--nodes should be 1 or greater",
-				)
-			}
-			c, err := common.NewDockerClient()
-			if err != nil {
-				return err
+				out.Die("--nodes should be 1 or greater")
 			}
+			c, err := common.NewDockerClient(cmd.Context())
+			out.MaybeDie(err, "unable to create docker client: %v", err)
 			defer c.Close()
 
 			configKvs := collectFlags(os.Args, "--set")
+			isRestarted, err := startCluster(c, nodes, checkBrokers, retries, image, pull, configKvs)
+			out.MaybeDieErr(common.WrapIfConnErr(err))
 
-			return common.WrapIfConnErr(startCluster(
-				c,
-				nodes,
-				checkBrokers,
-				retries,
-				image,
-				pull,
-				configKvs,
-			))
+			if noProfile || isRestarted {
+				return
+			}
+
+			dockerNodes, err := renderClusterInfo(c)
+			out.MaybeDie(err, "unable to render cluster info: %v; you may run 'rpk container status' to retrieve the cluster info", err)
+
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			y, err := cfg.ActualRpkYamlOrEmpty()
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			err = common.CreateProfile(fs, c, y)
+			if err == nil {
+				fmt.Printf("\nCreated %q profile.\n", common.ContainerProfileName)
+				renderClusterInteract(dockerNodes, true)
+				return
+			}
+			if errors.Is(err, common.ErrContainerProfileExists) {
+				fmt.Printf("Unable to create a profile for the rpk container: %v; you may delete it and create a new one running 'rpk profile create --from-rpk-container'\n", err)
```

Comment:
We need to say that the name is here since we don't output it anywhere really. What do you think about,

```
Unable to create a profile for the rpk container: %v.

You can retry profile creation by running:
    rpk profile delete %s; rpk profile create --from-rpk-container
```
and then using err, common.ContainerProfileName as args
(it's a bit long but at least it's a bit less wide?)